### ISSUE TITLE
RuntimeError: cuda runtime error

### ISSUE THREAD

**Issue by shanshuo** [sunnyxiaohu/R-C3D.pytorch#11]
Hi, thanks for your work. When I run `trainval_net.py`, I met cuda runtime error as follows. I use one 1080 Ti GPU. It seems that the error is related to `tdcnn.py` line 75, could you please help me to solve this issue? Thank you.

```
/opt/conda/conda-bld/pytorch_1524584710464/work/aten/src/THC/THCTensorScatterGather.cu:97: void THCudaTensor_gatherKernel(TensorInfo<Real, IndexType>, TensorInfo<Real, IndexType>, TensorInfo<long, IndexType>, int, IndexType) [with IndexType = unsigned int, Real = float, Dims = 3]: block: [0,0,0], thread: [127,0,0] Assertion `indexValue >= 0 && indexValue < src.sizes[dim]` failed.
THCudaCheck FAIL file=/opt/conda/conda-bld/pytorch_1524584710464/work/aten/src/THC/generic/THCTensorScatterGather.cu line=70 error=59 : device-side assert triggered
Traceback (most recent call last):
  File "/home/shuo/opt/pycharm-community-2017.3.4/helpers/pydev/pydevd.py", line 1668, in <module>
    main()
  File "/home/shuo/opt/pycharm-community-2017.3.4/helpers/pydev/pydevd.py", line 1662, in main
    globals = debugger.run(setup['file'], None, None, is_module)
  File "/home/shuo/opt/pycharm-community-2017.3.4/helpers/pydev/pydevd.py", line 1072, in run
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "/home/shuo/opt/pycharm-community-2017.3.4/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/home/shuo/code/R-C3D.pytorch/trainval_net.py", line 287, in <module>
    rois_label = tdcnn_demo(video_data, gt_twins)
  File "/home/shuo/anaconda3/lib/python3.6/site-packages/torch/nn/modules/module.py", line 491, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/shuo/anaconda3/lib/python3.6/site-packages/torch/nn/parallel/data_parallel.py", line 112, in forward
    return self.module(*inputs[0], **kwargs[0])
  File "/home/shuo/anaconda3/lib/python3.6/site-packages/torch/nn/modules/module.py", line 491, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/shuo/code/R-C3D.pytorch/lib/model/tdcnn/tdcnn.py", line 75, in forward
    twin_pred_select = torch.gather(twin_pred_view, 1, rois_label.view(rois_label.size(0), 1, 1).expand(rois_label.size(0), 1, 2))
RuntimeError: cuda runtime error (59) : device-side assert triggered at /opt/conda/conda-bld/pytorch_1524584710464/work/aten/src/THC/generic/THCTensorScatterGather.cu:70
terminate called without an active exception
```

**Comment by yzheng97** [comment]
Refer to my [issue](https://github.com/sunnyxiaohu/R-C3D.pytorch/issues/10)

The num_classes must be set correctly.

**Comment by shanshuo** [comment]
Thanks. But the issue is not with `args.num_classes`. I run THUMOS14 and the class number is set to 21. I have figured out. The problem is with `detclasslist.txt`. It should be:
1 BaseballPitch
2 BasketballDunk
3 Billiards
4 CleanAndJerk
5 CliffDiving
6 CricketBowling
7 CricketShot
8 Diving
9 FrisbeeCatch
10 GolfSwing
11 HammerThrow
12 HighJump
13 JavelinThrow
14 LongJump
15 PoleVault
16 Shotput
17 SoccerPenalty
18 TennisSwing
19 ThrowDiscus
20 VolleyballSpiking

However I used the file provided by [THUMOS14 Development kit](http://crcv.ucf.edu/THUMOS14/eval_kit/THUMOS14_evalkit_20140818.zip), which is:
7 BaseballPitch
9 BasketballDunk
12 Billiards
21 CleanAndJerk
22 CliffDiving
23 CricketBowling
24 CricketShot
26 Diving
31 FrisbeeCatch
33 GolfSwing
36 HammerThrow
40 HighJump
45 JavelinThrow
51 LongJump
68 PoleVault
79 Shotput
85 SoccerPenalty
92 TennisSwing
93 ThrowDiscus
97 VolleyballSpiking
